File: CASE_Simple_Example_V4/Hardened/HAMR/hamr/ext-c/FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase/FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase.c
#include <FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase_api.h>
#include <FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase.h>
#include <ext.h>

// This file will not be overwritten so is safe to edit

Unit HAMR_SW_FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase_initialise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase.c", "", "HAMR_SW_FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase_initialise_", 0);

  // examples of api setter and logging usage

  DeclNewHAMR_SW_MapArray(t0);
  HAMR_SW_MapArray_example(SF &t0);
  api_put_FlyZones__HAMR_SW_FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase(SF &t0);

  api_logInfo__HAMR_SW_FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase(SF string("Example logInfo"));

  api_logDebug__HAMR_SW_FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase(SF string("Example logDebug"));

  api_logError__HAMR_SW_FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase(SF string("Example logError"));
}

Unit HAMR_SW_FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase_finalise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase.c", "", "HAMR_SW_FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase_finalise_", 0);
}

File: CASE_Simple_Example_V4/Hardened/HAMR/src/main/component/HAMR/SW/FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase.scala
// #Sireum

package HAMR.SW

import org.sireum._
import HAMR._

// This file will not be overwritten so is safe to edit
object FlyZoneDatabase_Impl_SW_FlyZones_FlyZoneDatabase {

  def initialise(api: FlyZoneDatabase_Impl_Initialization_Api): Unit = {
    // example api usage

    api.logInfo("Example info logging")
    api.logDebug("Example debug logging")
    api.logError("Example error logging")

    api.put_FlyZones(SW.MapArray.example())
  }


  def activate(api: FlyZoneDatabase_Impl_Operational_Api): Unit = { }

  def deactivate(api: FlyZoneDatabase_Impl_Operational_Api): Unit = { }

  def finalise(api: FlyZoneDatabase_Impl_Operational_Api): Unit = { }

  def recover(api: FlyZoneDatabase_Impl_Operational_Api): Unit = { }
}
